src/config/app-config-endpoints.html

```diff
@@ -1,11 +1,55 @@
 <script>
 
   AppEndpoints = {
-    protocol: 'https://',
-    subdomain: 'subdomain',
-    domain: '.domain.com'
+    apiBase: {
+      protocol: 'https:',
+      subdomain: 'subdomain',
+      apiSuffix: 'api',
+      domain: 'domain.com'
+    }
   };
 
+  function resolveApiBaseUrl() {
+    const config = AppEndpoints.apiBase;
+    let { protocol, port } = window.location;
+    let [ subdomain, ...domain] = window.location.hostname.split('.');
```

Comment:
let { protocol, port='' } = window.location;
const domainParts = window.location.hostname.split('.');

// Localhost
if (domainParts.includes('localhost')) {
  // We have 2 cases "localhost" and "subdomain.localhost"
  if (domainParts.length > 1) domainParts[0] = domainParts[0].concat(`-${config.apiSuffix}`);
  else domainParts.unshift(config.subdomain);

  // The last part will always be localhost since that's what determins the machine it's routed to, replace it
  domainParts[domainParts.length - 1] = config.domain;
  port = '';
  protocol = config.protocol;
} else {
  // This is all of our other environments: "sub.domain.com" or "domain.com", hopefully we can drop this at some point
  if (domainParts.length === 2) domainParts.unshift(config.apiSuffix);
  else domainParts[0] = domainParts[0].concat(`-${config.apiSuffx}`);
}

const domain = domainParts.join('.');
AppEndpoints.apiBaseUrl = `${protocol}//${domain}${port}`;